### PR TITLE
Update docs related to file level error suppression

### PR DESCRIPTION
**Pull request by doolio**
Fixes: #8364

### CODE REVIEW

File: docs/linter.md [hunk 1]
```diff
@@ -226,14 +226,15 @@ Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.
 """  # noqa: E501
 ```
 
-To ignore all violations across an entire file, add `# ruff: noqa` to any line in the file, like so:
+To ignore all violations across an entire file, add a `# ruff: noqa` comment on its own line
+anywhere in the file preferably towards the top, like so:
```
Comment (zanieb) [inline docs/linter.md:230]: Perhaps we could say "... add the line `# ruff: noqa` anywhere in the file..." which feels a bit clear than "add a comment on its own line"? Perhaps with another clarifying note at the end of the section saying "Global noqa comments must be on their own line to disambiguate from comments which ignore violations on a single line"?

**Comment by zanieb** [comment]
Thank you so much for taking the time to contribute! I'm hoping we can make it a bit clearer, lmk what you think!

File: docs/linter.md (hunk 1 above)
Comment (doolio) [reply docs/linter.md:230]: From my admittedly simple tests (as I'm just learning to program) even the `# ruff: noqa` comment must be on its own line. If it is an inline comment then errors are not ignored. See

image

(where the error in lines 2 and 4 is `E741 Ambiguous variable name: 'O'`, although I don't understand why it is only highlighted on the 'O' and not the 'X' in line 1)

vs

image

Comment (zanieb) [reply docs/linter.md:230]: Oh that shouldn't conflict with what I'm suggesting. Ignoring violations on a single line is "add `# noqa` or `# noqa: RULE` to the line" whereas a global violation ignore is adding a new line with `# ruff: noqa`

Comment (doolio) [reply docs/linter.md]: I see. I've pushed a new comment to reword the docs as you suggest. Please let me know if you I should make further changes. Should I squash these changes into one commit and rebase onto the latest commit on `main`?

**Review by MichaReiser — APPROVED** [review]

**Review by charliermarsh — APPROVED** [review]
Will defer to @zanieb to merge.

File: docs/linter.md (hunk 1 above)
Comment (zanieb) [reply docs/linter.md]: We squash merge so you don't need to do anything with your commit.

I can take a look at that error later today!

**Comment by doolio** [comment]
> [Merge branch 'astral-sh:main' into docs-file-level-noqa](https://github.com/astral-sh/ruff/pull/8366/commits/1b869be7866a82accb95616ac160441b676630f5)

I did this in error.

**Comment by zanieb** [comment]
You can undo the commit with `git reset --hard HEAD^` then push with `git push --force` — but there's also no need merging `main` has no detrimental effect.

**Comment by doolio** [comment]
Thanks. I was concerned I would have screwed up the history for the merge into `main`. I've done as you suggested if for no reason other than it was a good learning exercise for me!

File: docs/linter.md (hunk 1 above)
Comment (zanieb) [reply docs/linter.md]: re the ambiguous name, "I" and "O" are the only names that are considered ambiguous https://docs.astral.sh/ruff/rules/ambiguous-variable-name/

Comment (doolio) [reply docs/linter.md]: Ah, interesting. Thank you for taking the time to look into this.

**Pull request merged**